core/test/mpi/distributed/solver/multigrid.cpp

```diff
@@ -0,0 +1,212 @@
+// SPDX-FileCopyrightText: 2017 - 2024 The Ginkgo authors
+//
+// SPDX-License-Identifier: BSD-3-Clause
+
+#include <gtest/gtest.h>
+
+
+#include <ginkgo/config.hpp>
+#include <ginkgo/core/distributed/matrix.hpp>
+#include <ginkgo/core/distributed/partition.hpp>
+#include <ginkgo/core/distributed/vector.hpp>
+#include <ginkgo/core/solver/cg.hpp>
+#include <ginkgo/core/solver/multigrid.hpp>
+#include <ginkgo/core/stop/iteration.hpp>
+
+
+#include "core/test/utils.hpp"
+
+
+namespace {
+
+
+using value_type = float;
+using local_index_type = int;
+using global_index_type = int;
+using dist_mtx_type =
+    gko::experimental::distributed::Matrix<value_type, local_index_type,
+                                           global_index_type>;
+using dist_vec_type = gko::experimental::distributed::Vector<value_type>;
+
+
+class DummyLinOp : public gko::EnableLinOp<DummyLinOp>,
```

Comment:
I guess that is needed to construct a DistributedMatrix without fully constructing `local_mtx` and `non_local_mtx` but getting access to the size. Isn't that the same as CustomLinOp? Also if it is implemented at several places maybe we should have a common implementation and reuse it.

Reply:
partially. they are not the same. the dummy class usually only do the minimal thing for the test. It's kind of trade-off in my mind. but I would say collecting them into a common type is a bit out of this pr scope